controller/front-end/sites.js

```diff
@@ -90,6 +92,172 @@ function initSitesController(dashboard, router) {
 			return next(error);
 		}
 	});
+	// List of all URLs associated with site
+	router.get('/sites/:siteId/urls', requirePermission('read'), async (request, response) => {
+		response.render('template/sites/urls', {
+			site: await Site.fetchOneById(request.params.siteId)
+		});
+	});
+
+	// Display the new URLs page
+	router.get('/sites/:siteId/urls/new', requirePermission('write'), async (request, response) => {
+		try {
+			const site = await Site.fetchOneById(request.params.siteId);
+			response.render('template/sites/new-url', {
+				site: site.serialize(),
+				urls: (await Url.fetchBySiteId(site.id)).serialize(),
+				form: {
+					url: {
+						created: request.flash.get('form.url.created')
+					}
+				}
+			});
+		} catch (error) {
+			return error;
+		}
+	});
+
+	// Add a new URL for site
+	router.post('/sites/:siteId/urls/new', requirePermission('write'), express.urlencoded({extended: false}), async (request, response, next) => {
+		try {
+
+			// TODO check Pa11y config for JSON errors
+			// and pass into the create method
+
+			const site = await Site.fetchOneById(request.params.siteId);
+			// Attempt to create a URL
+			await Url.create({
+				site_id: request.params.siteId,
+				name: request.body.name,
+				address: request.body.address,
+				pa11y_config: {}
+			});
+
+			// Redirect to the site page
+			request.flash.set('form.url.created', {});
+			response.redirect(`/sites/${site.get('id')}`);
+
+		} catch (error) {
+			if (error.name === 'ValidationError') {
+				return response.status(400).render('template/sites/new-url', {
+					form: {
+						url: {
+							site_id: request.body.site_id,
+							name: request.body.name,
+							address: request.body.address,
+							pa11y_config: request.body.pa11yConfig,
+							errors: error.details.map(detail => detail.message)
+						}
+					}
+				});
+			}
+			return next(error);
+		}
+	});
+
+	// Display the URL edit page
+	router.get('/sites/:siteId/urls/:urlId/edit', requirePermission('write'), async (request, response, next) => {
+		try {
+			const url = await Url.fetchOneById(request.params.urlId);
+			if (!url) {
+				return next();
```

Comment:
Is the `!url` check required? Does `Url.fetchOneById` not throw an exception if nothing was found?